packages/time-series-chart/src/Chart.tsx

```diff
@@ -0,0 +1,203 @@
+import * as React from "react";
+import { LineChart, ChartProps } from "./LineChart";
+import { options, series } from "./SampleData";
+import { ComponentController, DataSet, ColumnType } from "@dashbuilder-js/component-api";
+import { useState, useEffect, useCallback } from "react";
+import { Options, SingleSeries } from "./Data";
+
+// Default Values
+const NOT_ENOUGH_COLUMNS_MSG =
+  "Time series component expects at least 2 columns: category(LABEL or TEXT or NUMBER or DATE) and one or more series(NUMBER).";
+const FIRST_COLUMN_INVALID_MSG = "Wrong type for first column, it should be either LABEL, TEXT, NUMBER or DATE.";
+const SECOND_COLUMN_INVALID_MSG = "Wrong type for second column, it should be NUMBER.";
+const NOT_ENOUGH_COLUMNS_MSG_1 =
+  "Time series component expects 3 columns: category(LABEL or TEXT or NUMBER or DATE), series(TEXT) and values(NUMBER).";
+const FIRST_COLUMN_INVALID_MSG_1 = "Wrong type for first column, it should be either LABEL, TEXT, NUMBER or DATE.";
+const SECOND_COLUMN_INVALID_MSG_1 = "Wrong type for second column, it should be TEXT.";
+const THIRD_COLUMN_INVALID_MSG_1 = "Wrong type for third column, it should be NUMBER.";
+
+enum Params {
+  FLAG = "flag",
+  CHARTNAME = "chartName"
+}
+
+enum AppStateType {
+  ERROR = "Error",
+  INIT = "Initializing",
+  LOADING_COMPONENT = "Loading Component",
+  LOADED_COMPONENT = "Loaded Component",
+  FINISHED = "Finished loading"
+}
+
+interface AppState {
+  state: AppStateType;
+  processesoptions: Options;
+  processesseries: Array<SingleSeries>;
+  configurationIssue: string;
+  message?: string;
+}
+
+const validateParams = (params: Map<string, string | number>): string | undefined => {
+  if (!params.get(Params.FLAG)) {
+    return "Flag is required.";
+  }
+  if (!params.get(Params.CHARTNAME)) {
+    return "Chart name is required.";
+  }
+};
+
+const validateDataSet = (ds: DataSet, flag: number): string | undefined => {
+  if (flag === 0) {
+    if (ds.columns.length < 2) {
+      return NOT_ENOUGH_COLUMNS_MSG;
+    }
+    if (
+      ds.columns[0].type !== ColumnType.LABEL &&
+      ds.columns[0].type !== ColumnType.TEXT &&
+      ds.columns[0].type !== ColumnType.NUMBER &&
+      ds.columns[0].type !== ColumnType.DATE
+    ) {
+      return FIRST_COLUMN_INVALID_MSG;
+    }
+    if (ds.columns[1].type !== ColumnType.NUMBER) {
+      return SECOND_COLUMN_INVALID_MSG;
+    }
+  }
+  if (flag === 1) {
+    if (ds.columns.length < 3) {
```

Comment:
For clarity I also suggest you to extract the validation for transposed dataset and not transposed in different methods:
```
if(transposed) {
     return validateTransposedDataSet(dataSet);
}
else {
     // do the validation for not transposed datasets
}
```